gtsam/hybrid/HybridLookupDAG.h

```diff
@@ -0,0 +1,119 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file HybridLookupDAG.h
+ * @date Aug, 2022
+ * @author Shangjie Xue
+ */
+
+#pragma once
+
+#include <gtsam/discrete/DiscreteDistribution.h>
+#include <gtsam/discrete/DiscreteLookupDAG.h>
+#include <gtsam/hybrid/HybridConditional.h>
+#include <gtsam/hybrid/HybridValues.h>
+#include <gtsam/inference/BayesNet.h>
+#include <gtsam/inference/FactorGraph.h>
+
+#include <boost/shared_ptr.hpp>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace gtsam {
+
+/**
+ * @brief HybridLookupTable table for max-product
+ *
+ * Similar to DiscreteLookupTable, inherits from hybrid conditional for
+ * convenience. Is used in the max-product algorithm.
+ */
+class GTSAM_EXPORT HybridLookupTable : public HybridConditional {
+ public:
+  using Base = HybridConditional;
+  using This = HybridLookupTable;
+  using shared_ptr = boost::shared_ptr<This>;
+  using BaseConditional = Conditional<DecisionTreeFactor, This>;
+
+  /**
+   * @brief Construct a new Hybrid Lookup Table object form a HybridConditional.
+   *
+   * @param conditional input hybrid conditional
+   */
+  HybridLookupTable(HybridConditional& conditional) : Base(conditional){};
+
+  /**
+   * @brief Calculate assignment for frontal variables that maximizes value.
+   * @param (in/out) parentsValues Known assignments for the parents.
+   */
+  void argmaxInPlace(HybridValues* parentsValues) const;
+};
+
+/** A DAG made from hybrid lookup tables, as defined above. Similar to
+ * DiscreteLookupDAG */
+class GTSAM_EXPORT HybridLookupDAG : public BayesNet<HybridLookupTable> {
```

Comment:
The docs for this class are severely lacking. @xsj01 you have to be better than just copy-pasting the same docstrings, since now the motivation for this class is lost.